File: Interview practice/2022/6_14-Mark_and_Toys.js
/*
https://www.hackerrank.com/challenges/mark-and-toys/problem?isFullScreen=true&h_l=interview&playlist_slugs%5B%5D=interview-preparation-kit&playlist_slugs%5B%5D=sorting
*/
let array = [1, 12, 5, 111, 200, 1000, 10];
let budget = 50;
maximumToys(array, budget);

function maximumToys(prices, k) {
    let count = 0;
    let total = 0;
    prices = prices.sort(function(a, b){ return a - b; });
    console.log(prices);
    while(total <= k){
        console.log("total before : " + total);
        total += prices[count];
        console.log("total after : " + total)
        if(total > k){
            break;
        }
        count++;
        console.log("count is now : " + count)
    }

    return count;
}

File: Interview practice/2022/6_14-Mark_and_Toys.cs
/*
https://www.hackerrank.com/challenges/mark-and-toys/problem?isFullScreen=true&h_l=interview&playlist_slugs%5B%5D=interview-preparation-kit&playlist_slugs%5B%5D=sorting
*/
using System;

class Result{

    // static void Main()
    // {
    //     List<int> array = {1, 12, 5, 111, 200, 1000, 10};
    //     int budget = 50;

    //     maximumToys(array, budget);
    // }

    public static int maximumToys(List<int> prices, int k)
    {
        int count = 0;
        int total = 0;
        
        prices.Sort();
        while(total <= k){
            total += prices[count];
            if(total > k)
            {
                break;
            }
            count++;
        }
        return count;
    }
    
}